sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1709,26 +1695,61 @@ protected void afterCreating() {
             .getByResourceGroupAsync(this.resourceGroupName(), this.name());
     }
 
+
     // Helpers
     //
+    private void adjustProfileForFlexibleMode() {
+        if (this.orchestrationMode() == OrchestrationMode.FLEXIBLE) {
+            if (this.innerModel().virtualMachineProfile().networkProfile().networkInterfaceConfigurations() != null) {
+                this.innerModel().virtualMachineProfile().networkProfile().networkInterfaceConfigurations().forEach(virtualMachineScaleSetNetworkConfiguration -> {
+                    if (virtualMachineScaleSetNetworkConfiguration.ipConfigurations() != null) {
+                        virtualMachineScaleSetNetworkConfiguration.ipConfigurations().forEach(virtualMachineScaleSetIpConfiguration -> {
+                            // this property is not allowed to appear when creating vmss in flexible mode, though it's defined in the swagger file
+//                            if (CoreUtils.isNullOrEmpty(virtualMachineScaleSetIpConfiguration.loadBalancerInboundNatPools())) {
+                            virtualMachineScaleSetIpConfiguration.withLoadBalancerInboundNatPools(null);
+//                            }
+                        });
+                    }
+                });
+            }
+            this.innerModel()
+                // upgradePolicy is not supported in flexible vmss
+                .withUpgradePolicy(null)
+                .virtualMachineProfile().networkProfile()
+                // NetworkApiVersion must be specified when creating in flexible mode
+                .withNetworkApiVersion(NetworkApiVersion.TWO_ZERO_TWO_ZERO_ONE_ONE_ZERO_ONE);
+        }
+    }
+
+    private void initVMProfileIfNecessary() {
+        if (this.innerModel().virtualMachineProfile() == null) {
+            this.innerModel().withVirtualMachineProfile(initDefaultVMProfile());
+        }
+        this.profileNotSet = false;
+    }
+
+    private VirtualMachineScaleSetVMProfile initDefaultVMProfile() {
+        VirtualMachineScaleSetImpl impl = (VirtualMachineScaleSetImpl) this.manager()
+            .virtualMachineScaleSets()
+            .define(this.name());
+        if (this.orchestrationMode() == OrchestrationMode.FLEXIBLE) {
+            // orchestration mode is flexible and profile is null, so fault domain count must have been set
+            impl.withFlexibleOrchestrationMode(this.innerModel().platformFaultDomainCount());
```

Comment:
Could `platformFaultDomainCount()` be `null`?

Reply:
`platformFaultDomainCount` is mandatory when creating Flexible vmss, so it can't be `null`. But I'll add a protection in case service decides that it can.

Reply:
Added!